mailbox/api/src/test/java/org/apache/james/mailbox/MailboxManagerTest.java

```diff
@@ -2692,4 +2692,74 @@ void getMessagesShouldNotIncludeAttachmentInformationWhenNone() throws Exception
                 .satisfies(Throwing.consumer(messageResult -> assertThat(messageResult.hasAttachments()).isFalse()));
         }
     }
+
+    @Nested
+    class RightTests {
+        @BeforeEach
+        void setUp() {
+            session = mailboxManager.createSystemSession(USER_1);
+        }
+
+        @Test
+        void hasRightShouldThrowOnUnknownMailbox() {
+            assertThatThrownBy(() -> mailboxManager.hasRight(
+                    MailboxPath.forUser(USER_1, "notFound"),
+                    MailboxACL.Right.Administer,
+                    session))
+                .isInstanceOf(MailboxNotFoundException.class);
+        }
+
+        @Test
+        void listRightsShouldThrowOnUnknownMailbox() {
+            assertThatThrownBy(() -> mailboxManager.listRights(
+                    MailboxPath.forUser(USER_1, "notFound"),
+                    session))
+                .isInstanceOf(MailboxNotFoundException.class);
+        }
+
+        @Test
+        void myRightsShouldThrowOnUnknownMailbox() {
+            assertThatThrownBy(() -> mailboxManager.myRights(
+                    MailboxPath.forUser(USER_1, "notFound"),
+                    session))
+                .isInstanceOf(MailboxNotFoundException.class);
+        }
+
+        @Test
+        void listRightsForEntryShouldThrowOnUnknownMailbox() {
+            assertThatThrownBy(() -> mailboxManager.listRights(
+                    MailboxPath.forUser(USER_1, "notFound"),
+                    MailboxACL.EntryKey.createUserEntryKey(USER_2),
+                    session))
+                .isInstanceOf(MailboxNotFoundException.class);
+        }
+
+        @Test
+        void setRightsShouldNotThrowOnUnknownMailbox() {
+            assertThatCode(() -> mailboxManager.setRights(
+                    MailboxPath.forUser(USER_1, "notFound"),
+                    MailboxACL.EMPTY,
+                    session))
+                .doesNotThrowAnyException();
```

Comment:
Shouldn't that test fail? In `StoreRightManager` it seems you throw a `MailboxNotFoundException` in case you can't get it?

Reply:
I don't want to affect todays behaviour (that might very well be wrong) just avoid the NPE.

If you are not happy with this I let you open a ticket, and a PR to fix this.

Reply:
No I think we misunderstood... And I didn't look at it good enough first, forgot there was two `setRights`, one with `MailboxPath` an other with `MailboxId`

The code for `MailboxPath` didn't change, thus this test is correct. However you changed the one with `MailboxId` to throw a `MailboxNotFoundException` in case it does not exist, and I believe you miss a test about it then?

Reply:
Pushed a fixup, maybe would help understand my point. Feel free to keep it or not if you think it's a mistake